lib/puppet/provider/grafana_dashboard/grafana.rb

```diff
@@ -132,15 +133,15 @@ def save_dashboard(dashboard)
     data = {
       dashboard: dashboard.merge('title' => resource[:title],
                                  'id' => @dashboard ? @dashboard['id'] : nil,
-                                 'uid' => slug,
+                                 'uid' => @dashboard ? @dashboard['uid'] : nil,
```

Comment:
Is that nil for grafana to auto-generate it?

Reply:
yes, according to https://grafana.com/docs/grafana/latest/http_api/dashboard/

Reply:
Thanks for confirming